flytekit/core/promise.py

```diff
@@ -303,6 +365,8 @@ def __init__(self, var: str, val: Union[NodeOutput, _literals_models.Literal]):
         self._var = var
         self._promise_ready = True
         self._val = val
+        self._ref = None
```

Comment:
Is there a reason you added this @ByronHsu?  This is causing the linting errors (I think).

Reply:
well this and the Optional below.

Reply:
I workaround this by disabling the type check for `ref`. Changing to `Optional[NodeOutput]` causes so many errors on other files, so I think disabling it might be easier.